akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala

```diff
@@ -64,22 +64,20 @@ class ClientServerSpec extends AnyWordSpec with Matchers with BeforeAndAfterAll
   "The low-level HTTP infrastructure" should {
 
     "properly bind a server" in {
-      val (hostname, port) = SocketUtil.temporaryServerHostnameAndPort()
       val probe = TestSubscriber.manualProbe[Http.IncomingConnection]()
-      val binding = Http().bind(hostname, port).to(Sink.fromSubscriber(probe)).run()
+      val binding = Http().bind("0.0.0.0", 0).to(Sink.fromSubscriber(probe)).run()
       val sub = probe.expectSubscription() // if we get it we are bound
-      Await.result(binding, 1.second.dilated)
+      Await.result(binding, 1.second.dilated).unbind()
       sub.cancel()
     }
 
     "properly bind a server with a default port set via settings" in {
-      val (hostname, port) = SocketUtil.temporaryServerHostnameAndPort()
       val probe = TestSubscriber.manualProbe[Http.IncomingConnection]()
-      val settings = ServerSettings(system).withDefaultHttpPort(port)
-      val binding = Http().bind(hostname, settings = settings).to(Sink.fromSubscriber(probe)).run()
+      // not really testing anything here, problem is that it is hard to find an unused port otherwise
+      val settings = ServerSettings(system).withDefaultHttpPort(0)
+      val binding = Http().bind("0.0.0.0", settings = settings).to(Sink.fromSubscriber(probe)).run()
       val sub = probe.expectSubscription() // if we get it we are bound
-      val address = Await.result(binding, 1.second.dilated).localAddress
-      address.getPort shouldEqual port
+      Await.result(binding, 1.second.dilated).unbind()
```

Comment:
I guess we should check we did *not* bind to port 80 here then?

Reply:
Hmm, yes an ephemeral port range shouldn't be in that range. On the other hand, the test user shouldn't be able to bind ephemeral ports as well, so I'm not sure we win something in practice here.

Reply:
added